runtime/lua/vim/lsp/util.lua

```diff
@@ -1353,6 +1353,39 @@ function M.character_offset(buf, row, col)
   return str_utfindex(line, col)
 end
 
+-- for use with `formatexpr` if called without parms
+-- @param start_line 1-indexed line
+-- @param end_line 1-indexed line
+-- @param timeout_ms optional
+function M.formatexpr(start_line, end_line, timeout_ms)
+  if not start_line or not end_line then
+    if vim.fn.mode() == 'i' or vim.fn.mode() == 'R' then
+      -- `formatexpr` is also called when exceding
+      -- `textwidth` in insert mode
+      -- fall back to internal formatting
+      return 1
+    end
+    start_line = vim.v.lnum
+    end_line = start_line + vim.v.count - 1
+  end
+  if start_line > 0 and end_line > 0 then
+    local params = {
+      textDocument = { uri = vim.uri_from_bufnr(0) };
+      range = {
+        start = { line = start_line - 1; character = 0; };
+        ["end"] = { line = end_line - 1; character = 0; };
+      };
+    };
```

Comment:
does `make_position_params` or other make this possible to do more easily? I don't remember (it's fine if you don't know, I can follow up w/ it when I come back to this later).

Reply:
Don't think so as `make_position_parms` uses the cursor position which is not relevant here, so we still have to change the end and start.